sycl/source/detail/device_impl.cpp

```diff
@@ -895,8 +895,8 @@ bool device_impl::extOneapiCanCompile(
     // Get the shared_ptr to this object from the platform that owns it.
     std::shared_ptr<device_impl> Self = MPlatform->getOrMakeDeviceImpl(MDevice);
     return sycl::ext::oneapi::experimental::detail::
-        is_source_kernel_bundle_supported(Language,
-                                          std::vector<DeviceImplPtr>{Self});
+        is_source_kernel_bundle_supported(
+            Language, std::vector<DeviceImplPtr>{std::move(Self)});
```

Comment:
Please drop this, it'll be addressed in https://github.com/intel/llvm/pull/18270